spec/ref_arch_setup/bolt_helper_spec.rb

```diff
@@ -240,6 +241,147 @@
+      end
+    end
+
+    context "when bolt fails" do
+      expected_output = "No Good"
+      expected_status = 1
+
+      it "raises an error" do
+        error = "ERROR: bolt plan failed!"
+        expect(RefArchSetup::BoltHelper).to receive(:params_to_string) \
+          .with(params).and_return(params_str)
+        expect(RefArchSetup::BoltHelper).to receive(:`)\
+          .with(@expected_command).and_return(expected_output)
+        `(exit #{expected_status})`
+        expect($?).to receive(:success?).and_return(false) # rubocop:disable Style/SpecialGlobalVars
+        allow(RefArchSetup::BoltHelper).to receive(:puts)
+        expect { RefArchSetup::BoltHelper.run_plan_with_bolt(plan, params, nodes) }
+          .to raise_error(error)
+      end
+      it "outputs informative messages and raises an error" do
+        error = "ERROR: bolt plan failed!"
+        expect(RefArchSetup::BoltHelper).to receive(:params_to_string) \
+          .with(params).and_return(params_str)
+        expect(RefArchSetup::BoltHelper).to receive(:`)\
+          .with(@expected_command).and_return(expected_output)
+        `(exit #{expected_status})`
+        expect($?).to receive(:success?).and_return(false) # rubocop:disable Style/SpecialGlobalVars
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Output was: #{expected_output}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Running: #{@expected_command}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts)\
+          .with("Exit status was: #{expected_status}")
+        expect { RefArchSetup::BoltHelper.run_plan_with_bolt(plan, params, nodes) }
+          .to raise_error(error)
+      end
+    end
+
+    context "when ssh user and password options are sent" do
+      before do
+        RefArchSetup::BoltHelper.bolt_options = bolt_user_opts
+        @expected_command_with_ssh = "#{@expected_command} #{bolt_user_string}"
+      end
+
+      it "passes the arguments to bolt" do
+        expected_output = "All Good"
+        expected_status = 0
+        expect(RefArchSetup::BoltHelper).to receive(:params_to_string) \
+          .with(params).and_return(params_str)
+        expect(RefArchSetup::BoltHelper).to receive(:`)\
+          .with(@expected_command_with_ssh).and_return(expected_output)
+        `(exit #{expected_status})`
+        expect($?).to receive(:success?).and_return(true) # rubocop:disable Style/SpecialGlobalVars
+        expect(RefArchSetup::BoltHelper).to receive(:puts)
+          .with("Running: #{@expected_command_with_ssh}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts)\
+          .with("Exit status was: #{expected_status}")
+        expect(RefArchSetup::BoltHelper).to receive(:puts).with("Output was: #{expected_output}")
+        RefArchSetup::BoltHelper.run_plan_with_bolt(plan, params, nodes)
+      end
+    end
+
+    context "when ssh private key is sent" do
```

Comment:
This is the same test as above.  Covers the same code too.  Both test that is something is set in bolt options it gets added to the command.

Reply:
Removed